app/core/utils.coco.coffee

```diff
@@ -3,6 +3,29 @@ slugify = _.str?.slugify ? _.string?.slugify # TODO: why _.string on client and
 isCodeCombat = true
 isOzaria = false
 
+getAnonymizedName = (shouldAnonymous, session) ->
+  if shouldAnonymous and me.get('_id').toString() != session.get('creator')
+    anonymizingUser(session.get('creator'))
+  else
+    session.get('creatorName') or 'Anonymous'
+
+getAnonymizingStatus = (league, supermodel) ->
+  anonymousPlayerName = false
+  if league and anonymousPlayerName = features.enableAnonymization
+    fetchAnonymous = $.get('/esports/anonymous/' + league)
+    supermodel.trackRequest(fetchAnonymous)
+    return new Promise((resolve, reject) ->
+      fetchAnonymous.then((res) =>
+        console.log('called resolve', res.anonymous)
+        resolve(res.anonymous)
+      ))
+  return new Promise (resolve, reject) ->
+    resolve(anonymousPlayerName)
+
+anonymizingUser = (user) ->
+  id = user?.id ? user
+  $.i18n.t('general.player') + ' ' + id.slice(19)
```

Comment:
The random hex string here is not really how players name themselves (should be more like `Player528` than `Player3ea`). I think we should convert it to a straight 3- or 4-digit number, and also hash it so that it's more random than the ObjectIds are. Easiest way: `parseInt(id.slice(21), 16)`.

However, this is pretty not-random, because you're basically just pulling from [the incrementing counter at the end of the ObjectId](https://www.mongodb.com/docs/manual/reference/method/ObjectId/). So if a bunch of players are created within a similar time, they'll have similar ids. Maybe this is ok, but you could avoid it with a simple hashing algorithm that pulls from some or all of the ObjectId:

```coffee
  # djb2 algorithm
  hashString: (str) ->
    (str.charCodeAt i for i in [0...str.length]).reduce(((hash, char) -> ((hash << 5) + hash) + char), 5381)  # hash * 33 + c

  Math.abs(hashString(id)) % 10000
```